token/program-2022-test/tests/transfer_hook.rs

```diff
@@ -821,6 +965,150 @@ async fn success_transfers_using_onchain_helper() {
         .unwrap();
 }
 
+#[tokio::test]
+async fn success_transfers_with_fee_using_onchain_helper() {
+    let authority = Pubkey::new_unique();
+    let program_id = Pubkey::new_unique();
+    let mint_a_keypair = Keypair::new();
+    let mint_a = mint_a_keypair.pubkey();
+    let mint_b_keypair = Keypair::new();
+    let mint_b = mint_b_keypair.pubkey();
+    let amount = 10;
+
+    let transfer_fee_config_authority = Keypair::new();
+    let withdraw_withheld_authority = Keypair::new();
+    let transfer_fee_basis_points = TEST_FEE_BASIS_POINTS;
+    let maximum_fee = TEST_MAXIMUM_FEE;
+
+    let swap_program_id = Pubkey::new_unique();
+    let mut program_test = setup_program_test(&program_id);
+    program_test.add_program(
+        "my_swap",
+        swap_program_id,
+        processor!(process_instruction_swap),
+    );
```

Comment:
This test isn't actually testing the new on-chain helper for `invoke_transfer_checked_with_fee`, since it's calling the swap program processor defined earlier in this file, which does a swap by calling `invoke_transfer_checked`. It's just invoking `TransferChecked` and debiting the base fee.

You can just create another processor with the new helpers to test them, though!

Reply:
Good catch, thank you!